Assets/Drag.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using Microsoft.MixedReality.Toolkit.UI;
+using UnityEngine;
+
+public class Drag : MonoBehaviour
+{
+    ManipulationHandler manipulationHandler;
+    Vector3 startingPosition;
+    GameObject clone;
+
+    private void Awake()
+    {
+        //Debug.Log("running awake");
+        startingPosition = transform.position;
+        manipulationHandler = GetComponent<ManipulationHandler>();
+        manipulationHandler.OnManipulationStarted.AddListener(startedMotion);
+        manipulationHandler.OnManipulationEnded.AddListener(stoppedMotion);
+    }
+
+    private void stoppedMotion(ManipulationEventData arg0)
+    {
+        if (Vector3.Distance(transform.position, startingPosition) < 2) // how far you can move the block while it still will snap back
```

Comment:
This should really be done with Unity events (https://www.youtube.com/watch?v=mC_yUEnGQXM) , you should be able to see `OnManipulationEnded` if the block is still in collision with the other or not